### ISSUE TITLE
Pre-Select filters

### ISSUE THREAD

**Issue by Fyrearm** [bullhorn/career-portal#180]
I would like to preselect a filter when first loading the portal.  What would be the best approach to do this?  I was thinking of adding a query string parameter but didn't know what areas of the code I would have to modify in order to do this. I would of course need to modify the query based on the current query parameter but I would also need to check the box. Or should I just check the box and trigger the click event?

Basically what I have done is add another filter called Division and we would like to preselect this Division filter based on which one of our Divisions the users is coming from. We would simply set a query string parameter on the link to the portal.

**Comment by bulbajackel** [comment]
Hello Fyrearm,

The query string parameter I can see working pretty well.  For checking the box, you can use `data-ng-checked` which would check the box relevant to your logic. We use data-ng-checked currently in the [sidebar template](https://github.com/bullhorn/career-portal/blob/master/src/app/sidebar/sidebar.html).

As far as using a query parameter, you will want to probably add the logic into the [search service](https://github.com/bullhorn/career-portal/blob/master/src/app/services/search.service.js) and the [sidebar controller](https://github.com/bullhorn/career-portal/blob/master/src/app/sidebar/sidebar.controller.js).

If you need the syntax for obtaining the query string param, we do this to add a source when applying on the portal in the [apply service.](https://github.com/bullhorn/career-portal/blob/master/src/app/services/apply.service.js#L72)  

All in all you will want to modify at least 3 locations

- The sidebar controller
- The Sidebar template
- The Search service

Please let me know if you have any additional questions.

**Comment by Fyrearm** [comment]
Thanks for the feedback.  I can run this jQuery at the command prompt and it works perfectly. However I'm having trouble understanding the lifecycle of the angular page loading.  Where would I put this in code to get it to fire at the correct point.

` angular.element('#MYQUERYID').triggerHandler('click');`

I'm not sure how to use data-ng-checked, but doesn't that just check the box and not fire data-ng-click?

**Comment by charlesabarnes** [comment]
You can possibly add in your logic for this in the constructor for the sidebar.  I am not too sure about utilizing jQuery though.  

The data-ng-checked is to define the criteria for it to be checked so if you did the logic elsewhere it would eliminate the need for the click.
